src/x509_scitokens_issuer/x509_scitokens_issuer.py

```diff
@@ -272,10 +276,21 @@ def token_issuer():
 
     creds = {}
     dn_cred = None
+    entry_num = 0
     for key, val in request.environ.items():
-        if key.startswith("GRST_CRED_AURI_"):
-            entry_num = int(key[15:]) # 15 = len("GRST_CRED_AURI_")
+        pattern = "GRST_CRED_AURI_"
+        if app.config.get("CMS", False):
+            pattern = "HTTP_CMS_AUTH"
```

Comment:
The `pattern` variable ought to be static within the for-loop.  Hoist it above.

Reply:
done